src/services/baseService.ts

```diff
@@ -19,24 +21,41 @@ export abstract class BaseService {
   ) {
     Guard.null(serverless);
 
+    this.setDefaultValues();
+
     this.baseUrl = "https://management.azure.com";
+    this.config = serverless.service as any;
     this.serviceName = serverless.service["service"];
     this.credentials = serverless.variables["azureCredentials"];
     this.subscriptionId = serverless.variables["subscriptionId"];
     this.resourceGroup = this.getResourceGroupName();
     this.deploymentName = serverless.service.provider["deploymentName"] || `${this.resourceGroup}-deployment`;
 
-    this.setDefaultValues();
-
     if (!this.credentials && authenticate) {
       throw new Error(`Azure Credentials has not been set in ${this.constructor.name}`);
     }
   }
 
-  public getResourceGroup(): string {
-    return this.resourceGroup;
+  public getRegion(): string {
+    return this.options.region || this.serverless.service.provider.region;
   }
   
+  public getStage(): string {
+    return this.options.stage || this.serverless.service.provider.stage;
+  }
+
+  public getFunctionAppName(): string {
+    return this.config.provider.functionApp
+      ? this.config.provider.functionApp.name
+      : `${this.config.provider.prefix}-${this.config.provider.region}-${this.config.provider.stage}-${this.config.service}`;
```

Comment:
Same as a above, would be cool to have a central place that captured this naming convention

Reply:
Each resource now exposes a a `getResourceName()` method that takes in the sls config and returns the resolved name based on naming convention.